File: app-mock/src/user/dto/update-user.dto.ts
import { IsString, IsInt, IsEmail, IsOptional } from 'class-validator';

export class UpdateUserDto {

    @IsString()
    @IsOptional()
    firstName: string;
  
    @IsString()
    @IsOptional()
    lastName: string;
  
    @IsInt()
    @IsOptional()
    age: number;
  
    @IsEmail()
    @IsOptional()
    email: string;

}

File: sap-mock/RestDTOs/UpdateUserDto.cs
using Newtonsoft.Json;

namespace sap_mock.RestDTOs {
    public class UpdateUserDto {

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public int Age { get; set; }
            
        public string Email { get; set; }
    }
}